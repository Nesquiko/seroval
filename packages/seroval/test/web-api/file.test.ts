import { describe, it, expect } from 'vitest';
import 'node-fetch-native/polyfill';
import {
  crossSerializeAsync,
  deserialize,
  fromJSON,
  serializeAsync,
  toJSONAsync,
} from '../../src';

describe('File', () => {
  describe('serializeAsync', () => {
    it('supports File', async () => {
      const example = new File(['Hello World'], 'hello.txt', {
        type: 'text/plain',
        lastModified: 1681027542680,
      });
      const result = await serializeAsync(Promise.resolve(example));
      expect(result).toMatchSnapshot();
      const back = await deserialize<Promise<File>>(result);
      expect(back).toBeInstanceOf(File);
      expect(await back.text()).toBe(await example.text());
      expect(back.type).toBe(example.type);
    });
  });
  describe('toJSONAsync', () => {
    it('supports File', async () => {
      const example = new File(['Hello World'], 'hello.txt', {
        type: 'text/plain',
        lastModified: 1681027542680,
      });
      const result = await toJSONAsync(Promise.resolve(example));
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = await fromJSON<Promise<File>>(result);
      expect(back).toBeInstanceOf(File);
      expect(await back.text()).toBe(await example.text());
      expect(back.type).toBe(example.type);
    });
  });
  describe('crossSerializeAsync', () => {
    it('supports File', async () => {
      const example = new File(['Hello World'], 'hello.txt', {
        type: 'text/plain',
        lastModified: 1681027542680,
      });
      const result = await crossSerializeAsync(Promise.resolve(example));
      expect(result).toMatchSnapshot();
    });
    describe('scoped', () => {
      it('supports File', async () => {
        const example = new File(['Hello World'], 'hello.txt', {
          type: 'text/plain',
          lastModified: 1681027542680,
        });
        const result = await crossSerializeAsync(Promise.resolve(example), { scopeId: 'example' });
        expect(result).toMatchSnapshot();
      });
    });
  });
});
