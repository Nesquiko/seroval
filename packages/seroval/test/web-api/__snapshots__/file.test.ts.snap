// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`File > crossSerializeAsync > scoped > supports File 1`] = `"((self.$R=self.$R||{})[\\"example\\"]=[],($R=>$R[0]=new File([$R[1]=new Uint8Array([72,101,108,108,111,32,87,111,114,108,100]).buffer],\\"hello.txt\\",{type:\\"text/plain\\",lastModified:1681027542680}))($R[\\"example\\"]))"`;

exports[`File > crossSerializeAsync > supports File 1`] = `"(self.$R=self.$R||[],$R[0]=new File([$R[1]=new Uint8Array([72,101,108,108,111,32,87,111,114,108,100]).buffer],\\"hello.txt\\",{type:\\"text/plain\\",lastModified:1681027542680}))"`;

exports[`File > serializeAsync > supports File 1`] = `"new File([new Uint8Array([72,101,108,108,111,32,87,111,114,108,100]).buffer],\\"hello.txt\\",{type:\\"text/plain\\",lastModified:1681027542680})"`;

exports[`File > toCrossJSONAsync > supports File 1`] = `"{\\"t\\":24,\\"i\\":0,\\"c\\":\\"text/plain\\",\\"m\\":\\"hello.txt\\",\\"f\\":{\\"t\\":21,\\"i\\":1,\\"s\\":[72,101,108,108,111,32,87,111,114,108,100]},\\"b\\":1681027542680}"`;

exports[`File > toJSONAsync > supports File 1`] = `"{\\"t\\":{\\"t\\":24,\\"i\\":0,\\"c\\":\\"text/plain\\",\\"m\\":\\"hello.txt\\",\\"f\\":{\\"t\\":21,\\"i\\":1,\\"s\\":[72,101,108,108,111,32,87,111,114,108,100]},\\"b\\":1681027542680},\\"f\\":16383,\\"m\\":[]}"`;
